input/mocks.go

```diff
@@ -0,0 +1,125 @@
+package input
+
+import (
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/stretchr/testify/mock"
+)
+
+// MockInput implements the `Input` interface and is used by other packages for
+// mock testing.
+type MockInput struct {
+	mock.Mock
```

Comment:
Def gotta use these more. More ergonomic (and less deadlock prone!) that using a series of custom internal channels to send/recv args or return values.

Reply:
yeah I use it a lot, especially in the [new payment lifecycle tests](https://github.com/lightningnetwork/lnd/pull/7503/commits/f3d435b7cc54064bdd5ced15e9da5b12b5968a60#diff-4963b0f520937f7e5a4aa825c3541d17af3514a2c7d8f94ce83ccbe4308b0739R692) where the control flow is easily handled.